src/aggregator/aggregator/options.go

```diff
@@ -387,6 +387,11 @@ type options struct {
 
 // NewOptions create a new set of options.
 func NewOptions() Options {
+	return NewOptionsWithClock(clock.NewOptions())
+}
+
+// NewOptionsWithClock create a new set of options using the supplied clock options.
+func NewOptionsWithClock(clockOpts clock.Options) Options {
```

Comment:
The default clock gets referenced during `initPools()` call below, before there's a chance to override it.

https://github.com/m3db/m3/blob/010e134e450cde7239870256ba921507ec223301/src/aggregator/aggregator/options.go#L424-L425

https://github.com/m3db/m3/blob/010e134e450cde7239870256ba921507ec223301/src/aggregator/aggregator/options.go#L827-L829

https://github.com/m3db/m3/blob/010e134e450cde7239870256ba921507ec223301/src/aggregator/aggregator/entry.go#L204-L209